lib/task/import/csvImportTask.class.php

```diff
@@ -996,18 +996,33 @@ public function execute($arguments = array(), $options = array())
         }
         else if ($path = $self->rowStatusVars['digitalObjectPath'])
         {
-          if (false === $content = file_get_contents($path))
+          $do = new QubitDigitalObject;
+          $do->usageId = QubitTerm::MASTER_ID;
+          $do->informationObject = $self->object;
+
+          // Don't create derivatives (reference, thumb)
+          if ($self->status['options']['skip-derivatives'])
           {
-            $this->log("Unable to read file: ".$path);
+            $do->createDerivatives = false;
           }
-          else
+
+          $do->assets[] = new QubitAsset($path);
+
+          try
           {
-            $do = new QubitDigitalObject;
-            $do->assets[] = new QubitAsset($path, $content);
-            $do->usageId = QubitTerm::MASTER_ID;
-            $do->informationObject = $self->object;
             $do->save($conn);
           }
+          catch (Exception $e)
+          {
+            // Echo error to STDOUT
+            $this->log($e->getMessage());
+
+            // Log error
+            if (sfConfig::get('sf_logging_enabled'))
+            {
+              sfContext::getInstance()->getLogger()->err($e->getMessage());
```

Comment:
Hmm, I got that snippet directly from [Listing 16-6 - Adding a Custom Log Message from Anywhere](http://librosweb.es/libro/symfony_1_2_en/chapter_16/logging.html), but it's from symfony 1.2 and there's no 1.4 version of the doc.  I'll try removing the conditional though, and see what happens. :)